File: autograder/services/templates/template.rb
# Copyright © 2026 Rutgers, the State University of New Jersey. All rights reserved except as defined by the Rutgers Non-Commercial License, included with this software.
# Ruby Execution Template
require 'open3'

packages_to_install = [] # REPLACED_BY_EXECUTOR

def log(msg, level)
  $stderr.puts "[#{level}] #{msg}"
end

unless packages_to_install.empty?
  log("Installing specific gems: #{packages_to_install.join(', ')}", "INFO")
  packages_to_install.each do |gem_name|
    begin
      if system("gem install --no-document #{gem_name}")
         log("✓ #{gem_name} installed", "INFO")
         $stderr.puts "CODEPOST_AUTO_INSTALL_SUCCESS: #{gem_name}"
      else
         log("Failed to install #{gem_name}", "ERROR")
      end
    rescue => e
      log("Error installing #{gem_name}: #{e}", "ERROR")
    end
  end
end

$stderr.puts "<<<RESULT>>>"

# USER CODE BEGINS
begin
  # FILLER_CODE
rescue => e
  $stderr.puts "#{e.class}: #{e.message}"
  $stderr.puts e.backtrace.join("\n")
  exit(1)
end

File: autograder/services/templates/template.php
// Copyright © 2026 Rutgers, the State University of New Jersey. All rights reserved except as defined by the Rutgers Non-Commercial License, included with this software.
<?php
// PHP Execution Template

$packages_to_install = []; // REPLACED_BY_EXECUTOR

function template_log($msg, $level) {
    fwrite(STDERR, "[$level] $msg\n");
}

if (!empty($packages_to_install)) {
    template_log("Installing packages: " . implode(', ', $packages_to_install), "INFO");
    // Composer require
    // Assuming composer is available globally
    foreach ($packages_to_install as $pkg) {
         exec("composer require --no-interaction " . escapeshellarg($pkg), $output, $return_var);
         if ($return_var === 0) {
             template_log("✓ $pkg installed", "INFO");
             fwrite(STDERR, "CODEPOST_AUTO_INSTALL_SUCCESS: $pkg\n");
         } else {
             template_log("Failed to install $pkg", "ERROR");
         }
    }
}

fwrite(STDERR, "<<<RESULT>>>\n");

// USER CODE BEGINS
// We use include logic or just paste code.
// Since PHP needs <?php tag, but we are inside one.
// The user code presumably starts with <?php ? or implies it?
// Usually user scripts have <?php. If we assume they do, we must handle it.
// If we paste it here, we should strip the opening tag or close ours.

// OPTION: Close PHP tag, then paste code (which opens it)
?>
# FILLER_CODE
